### ISSUE TITLE
DICOM Sequence Tags (SQ) handling

### ISSUE THREAD

**Issue by ebremer** [w3c/hcls-fhir-rdf#146]
[DICOM Sequence Tags (SQ)](https://dicom.nema.org/medical/dicom/current/output/html/part05.html#sect_7.5)

Seems straight forward enough to handle with an RDF Lists using blank nodes:
"The VR identified "SQ" shall be used for Data Elements with a Value consisting of a Sequence of zero or more Items, where each Item contains a set of Data Elements"

**Comment by ebremer** [comment]
Example, DICOM tag 30060080  is a VR of SQ (Sequence):  (List elements can be blank node, URN, or URI)
```
<urn:oid:aninstanceofaSOPInstanceUID>
        dcm:30060080  ( [ dcm:30060082  1;
                          dcm:30060084  1;
                          dcm:30060088  "Type:Soft, Range:*/*, Fill:0, Opacity:0.0, Thickness:1, LineThickness:2, read-only:false";
                          dcm:300600A4  [ dcm:vr  "CS" ];
                          dcm:300600A6  [ dcm:vr  "PN" ]
                        ]
                        [ dcm:30060082  2;
                          dcm:30060084  2;
                          dcm:30060088  "Type:Soft, Range:*/*, Fill:0, Opacity:0.0, Thickness:1, LineThickness:2, read-only:false";
                          dcm:300600A4  [ dcm:vr  "CS" ];
                          dcm:300600A6  [ dcm:vr  "PN" ]
                        ]
                        [ dcm:30060082  3;
                          dcm:30060084  3;
                          dcm:30060088  "Type:Soft, Range:*/*, Fill:0, Opacity:0.0, Thickness:1, LineThickness:2, read-only:false";
                          dcm:300600A4  [ dcm:vr  "CS" ];
                          dcm:300600A6  [ dcm:vr  "PN" ]
                        ]
                        [ dcm:30060082  4;
                          dcm:30060084  4;
                          dcm:30060088  "Type:Soft, Range:*/*, Fill:0, Opacity:0.0, Thickness:1, LineThickness:2, read-only:false";
                          dcm:300600A4  [ dcm:Value  ( "PTV" );
                                          dcm:vr     "CS"
                                        ];
                          dcm:300600A6  [ dcm:vr  "PN" ]
                        ]
                        [ dcm:30060082  5;
                          dcm:30060084  5;
                          dcm:30060088  "Type:Soft, Range:*/*, Fill:0, Opacity:0.0, Thickness:1, LineThickness:2, read-only:false";
                          dcm:300600A4  [ dcm:vr  "CS" ];
                          dcm:300600A6  [ dcm:vr  "PN" ]
                        ]
                      );
```

**Comment by detlef-grittner-sohard** [comment]
If URIs are always used instead of blank nodes, then it is possible to link the individual elements to triples in other namespaces reliably. 

In case of an item in a Structure Set ROI Sequence (3006,0020) one could link additional information about regions of interest to the ROI meta-data in the DICOM object.

**Comment by detlef-grittner-sohard** [comment]
Referencing an item representing an ROI  [(3006,0020)](https://dicom.nema.org/medical/dicom/current/output/html/part03.html#sect_C.8.8.5)
Originally this URL has been used for encoding an Item inside the ROI Sequence in an RT Struct:
`http://semantic-dicom.org/dcm#StructureSetROISequenceItem_99999.70989091007225463773458855903144372742_30060020.2`

with Erich Bremer's notation the URL looks more like this:
`<urn:oid:99999.70989091007225463773458855903144372742/30060020>`

This is a reference to the Name of the ROI in the DICOM domain:
`<urn:oid:99999.70989091007225463773458855903144372742/30060020/2> http://semantic-dicom.org/dcm#ATT30060026 "Heart"`

This is a reference in another graph using the item resource and link it to a volume of interest:
`<urn:oid:99999.70989091007225463773458855903144372742/30060020/2> http://semantic-dicom.org/rt#hasVOI http://some_voi`

The resource `http://some_voi` is already outside the DICOM domain and could contain information like algorithm used, the actual volume values, radiation to that volume etc.

**Comment by dbooth-boston** [comment]
URN syntax: https://www.rfc-editor.org/rfc/rfc8141#section-2  Because of the URN equivalence rules defined in [section 3](https://www.rfc-editor.org/rfc/rfc8141#section-3), we probably should only use the assigned-name part of the URN syntax, which now explicitly allows slash characters.

**Comment by dbooth-boston** [comment]
On second thought, in the RDF world two URNs with different fragments will be different identifiers, so I think it makes sense to put a list item number in the fragment part.

**Comment by dbooth-boston** [comment]
Discussed last week: https://www.w3.org/2025/03/20-hcls-minutes.html

**Comment by ebremer** [comment]
Proposed solution for consistent naming "detlefication" of sequence blank nodes entries:

Use URN for SOP instance UID with path to list item as fragment #.

Example - The following represents the URI for the third sequence item in the sequence specified by DICOM tag 3006,0020. 
```
<urn:oid:99999.70989091007225463773458855903144372742#30060020/2>
```
 The pattern above would be continued for sequences nested in sequences.

**Comment by dbooth-boston** [comment]
AGREED: to go ahead with this proposal https://www.w3.org/2025/04/03-hcls-minutes.html#d713
